scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -218,68 +218,47 @@ class FormatWriter(formatOps: FormatOps) {
 
         import org.scalafmt.config.TrailingCommas
 
-        val owner = owners(tok.right)
-        if (!runner.dialect.allowTrailingCommas ||
-          !TreeOps.isImporterOrDefnOrCallSite(owner)) {
-          sb.append(getWhitespace(0))
-          return
-        }
-        val isImport = owner.isInstanceOf[Importer]
-
-        val left = tok.left
-        val (skip, nextNonCommentTok) = nextNonCommentWithCount(tok)
-        val right = nextNonCommentTok.right
-        val isNewline =
-          Seq(curr, locations(math.min(i + skip, locations.length - 1)))
-            .exists(_.state.split.modification.isNewline)
-
-        // Scala syntax allows commas before right braces in weird places,
-        // like constructor bodies:
-        // def this() = {
-        //   this(1),
-        // }
-        // This code simply ignores those commas because it does not
-        // consider them "trailing" commas. It does not remove them
-        // in the TrailingCommas.never branch, nor does it
-        // try to add them in the TrainingCommas.always branch.
-        lazy val rightIsCloseDelim = right
-          .is[CloseParenOrBracket] || (right.is[T.RightBrace] && isImport)
-
-        initStyle.trailingCommas match {
-          // foo(
-          //   a,
-          //   b
-          // )
-          //
-          // Insert a comma after b
-          case TrailingCommas.always
-              if !left.is[T.Comma] &&
-                !left.is[T.Comment] &&
-                !left.is[T.LeftParen] && // skip empty parentheses
-                rightIsCloseDelim && isNewline =>
-            sb.append(',').append(getWhitespace(-1))
-
-          // foo(
-          //   a,
-          //   b,
-          // )
-          //
-          // Remove the comma after b
-          case TrailingCommas.never
-              if left.is[T.Comma] && rightIsCloseDelim && isNewline =>
-            sb.setLength(sb.length - 1)
-            sb.append(getWhitespace(1))
-
-          // foo(a, b,)
-          //
-          // Remove the comma after b
-          case _ if left.is[T.Comma] && rightIsCloseDelim && !isNewline =>
-            sb.setLength(sb.length - 1)
-
-          case _ => sb.append(getWhitespace(0))
+        val appendComma = runner.dialect.allowTrailingCommas &&
+          initStyle.trailingCommas == TrailingCommas.always && {
+          tok.left match {
+            case _: T.Comma | _: T.Comment | _: T.LeftParen | _: T.LeftBrace =>
+              // skip empty parens/braces
```

Comment:
This comment means:

if the left token is `{` or `(`, that means the parentheses/braces is empty, and if we add a comma there, the result would be something like

```scala
foo (,)
```

that's why we skip adding commas for empty parens/braces.

---

It's probably better to move this comment like this:

```scala
            case _: T.LeftParen | _: T.LeftBrace | // skip empty parens/braces
                 _: T.Comma | _: T.Comment =>
```

 So that people can easily understand we are skipping empty parens/braces by not adding commas if the left token is `(` or `{`.

Reply:
will clarify :) in fact, original code only checked for '(', so i am improvising here.

Reply:
done